### ISSUE TITLE
Bluespace bears infinite loop

### ISSUE THREAD

**Issue by skull132** [Aurorastation/Aurora.3#944]
```
[19:08:39] Process mob caught exception processing /mob/living/simple_animal/hostile/bear/spatial (the bluespace bear) (135,105,1): Maximum recursion level reached (perhaps there is an infinite loop)
To avoid this safety check, set world.loop_checks=0. in , line
```

**Comment by skull132** [comment]
Testing results:
```
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Edgar Fiddler
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 568.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Bill Green
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 569.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Edgar Fiddler
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 570.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Bill Green
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 571.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Edgar Fiddler
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 572.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Bill Green
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 573.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Edgar Fiddler
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 574.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Bill Green
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 575.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Edgar Fiddler
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 576.
[15:28:23] b3daVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Bill Green
[15:28:23] b3daVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:28:23] b3daVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:28:23] b3daVf DEBUG: TARGET: Called from LoseTarget()
[15:28:23] b3daVf DEBUG: TARGET: FindTarget() at iteration 577.
```

This is as far as my current stack trace results go. It seems to be MoveToTarget causing it, which is called by FoundTarget. Annnd FoundTarget itself is called by FindTarget, though I need to confirm real quick if that's the issue.

**Comment by skull132** [comment]
More exact information. The issue is that it keeps calling FoundTarget() in FindTarget() which then calls FindTarget() again. The resulting behaviour is that it cycles between two conditions in an if/elseif chain here:
```
	if (nearest_target)
		target_mob = nearest_target
		log_debug("TARGET: FoundTarget() called by FindTarget():230.")
		FoundTarget()
		return nearest_target
	else if (nearest_downed_target)
		target_mob = nearest_downed_target
		log_debug("TARGET: FoundTarget() called by FindTarget():236.")
		FoundTarget()
		return nearest_downed_target
```

Debug logs:
```
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Devon Hynes
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 230.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():230.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Eli Martins
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 231.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():236.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Devon Hynes
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 232.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():230.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Eli Martins
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 233.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():236.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Devon Hynes
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 234.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():230.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Eli Martins
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 235.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():236.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Devon Hynes
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 236.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():230.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Eli Martins
[15:41:08] bQfaVf DEBUG: TARGET: MoveToTarget called from FoundTarget().
[15:41:08] bQfaVf DEBUG: TARGET: LoseTarget() called in hostile.dm:78.
[15:41:08] bQfaVf DEBUG: TARGET: Called from LoseTarget()
[15:41:08] bQfaVf DEBUG: TARGET: FindTarget() at iteration 237.
[15:41:08] bQfaVf DEBUG: TARGET: FoundTarget() called by FindTarget():236.
[15:41:08] bQfaVf EMOTE: space bear/ : <B>The space bear</B> stares alertly at Devon Hynes
```

**Comment by NanakoAC** [comment]
FIXED BY #945
